File: js/fibonacci.js

const fib = (n) => {
    const arr = [0, 1];  // Starting from 0, 1 for the Fibonacci sequence
    let position = 2;
    while (position <= n) {
        arr.push(arr[position - 2] + arr[position - 1]);
        position += 1;
    }
    return arr[n];
};

console.log(fib(2));
console.log(fib(3));
console.log(fib(4));
console.log(fib(5));

File: python/fibonacci.py
# https://www.codewars.com/kata/53d40c1e2f13e331fc000c26/train/python
def fib(n):
    is_pos = abs(n) == n
    i = abs(n)
    arr = [0,1]
    while len(arr) < i + 1:
        arr.append(arr[-2] + arr[-1])
    return arr[i] if is_pos else arr[i] * -1

print(fib(0)) # 0
print(fib(1)) # 1
print(fib(2)) # 1
print(fib(3)) # 2
print(fib(4)) # 3
print(fib(5)) # 5
print(fib(6)) # 8
print(fib(7)) # 13
print(fib(8)) # 21
print(fib(-1)) # 1
